### PR TITLE
add more info around custom partition_size

### PR DESCRIPTION
**Pull request by samantehrani**
based on https://github.com/ArweaveTeam/arweave/blob/72783cb45aff78738f353f502468bf7732ee4135/apps/arweave/src/ar_storage_module.erl#L228

### CODE REVIEW

File: mining/mining-guide.md [hunk 1]
```diff
@@ -83,20 +83,21 @@ In order to mine in 2.6, your mining key needs to be present on your machine. If
...
-
-Mount your drives in the `[data_dir]/storage_modules` folder. E.g.,
+The first step is to create a folder inside `[data_dir]/storage_modules/` for each of the partisions that you intend to mine with. Partition folder names should use the following pattern: `storage_module[_partition_size]_[partition_index]_[your_mining_address]`. As you will see further, specifying `partition_size` is optional. The default partition size is 3.6 TB, as Arweave dataset is logically partitioned into collections of 3.6 TB "mining partitions". For any given `partition_size`, you should take into account around 10% metadata overhead (such as merkle proofs). Hence, for the default 3.6 TB `partition_size`, we recommend a 4 TB disk. The partitions are indexed sequentially starting from 0, with 0 being the very first 3.6 TB (or `partition_size`) worth of data stored on Arweave, and ranging to whatever the current mining partition count is. You can choose which mining partitions you store by indicating their `partition_index` in the folder name.
```
Comment (JamesPiechota) [inline mining/mining-guide.md:86]:
```suggestion
The Arweave dataset is logically partitioned into collections of 3.6 TB "mining partitions". You will store some or all of those mining partitions on your miner in "storage modules". Storage modules can be any size (smaller or larger than the default 3.6TB mining partition size), but many miners opt to align their storage modules with the Arweave partitions.

To setup your storage modules, the first step is to create a folder inside `[data_dir]/storage_modules/` for each of the storage modules that you intend to mine with. Storage module folder names should use the following pattern: `storage_module[_storage_module_size]_[storage_module_index]_[your_mining_address]`. The default storage module size is 3.6TB - in that case specifying `storage_module_size` is optional.

For any given `storage_module_size`, you should allow for an additional 10% metadata overhead (such as merkle proofs). This is why, for the default 3.6 TB `storage_module_size`, we recommend a 4 TB disk. The storage modules are indexed sequentially starting from 0, with 0 being the very first 3.6 TB (or `storage_module_size`) worth of data stored on Arweave, and ranging up to or beyond the current Arweave dataset size (`weave_size`). You can choose which mining partitions you store by indicating their `storage_module_index` in the folder name. 
```

Comment (JamesPiechota) [reply mining/mining-guide.md:86]: Sorry for all the edits. The distinction between storage module and partition is awkward but important... which I think is why I've put off updating the docs.

File: mining/mining-guide.md [hunk 2]
```diff
@@ -83,20 +83,21 @@ In order to mine in 2.6, your mining key needs to be present on your machine. If
...
+For example, to set up a storage module with the very first mining partition in the weave with the default 3.6 TB `partition_size` (packed with your mining address), create the folder such as `[data_dir]/storage_modules/storage_module_0_[your_mining_address]`. If you were to store 2 TB Partitions, create the folder such as: `[data_dir]/storage_modules/storage_module_2000000000000_0_[your_mining_address]`.
```
Comment (JamesPiechota) [inline mining/mining-guide.md:87]:
```suggestion
For example, to set up a storage module with the very first mining partition in the weave with the default 3.6 TB `storage_module_size` (packed with your mining address), create the folder such as `[data_dir]/storage_modules/storage_module_0_[your_mining_address]`. If you were to store 2 TB storage modules, create the folder such as: `[data_dir]/storage_modules/storage_module_2000000000000_0_[your_mining_address]`.
```

File: mining/mining-guide.md [hunk 3]
```diff
@@ -83,20 +83,21 @@ In order to mine in 2.6, your mining key needs to be present on your machine. If
...
 
+After creating the relevant folders for your chosen partitions, mount your drives onto them. E.g.,
 ```
 sudo mount /dev/sda [data_dir]/storage_modules/storage_module_0_[your_mining_address]
+sudo mount /dev/sda [data_dir]/storage_modules/storage_module_2000000000000_0_[your_mining_address] // custom 2 tb partition size
```
Comment (JamesPiechota) [inline mining/mining-guide.md:92]:
```suggestion
sudo mount /dev/sda [data_dir]/storage_modules/storage_module_2000000000000_0_[your_mining_address]
```

File: mining/mining-guide.md [hunk 4]
```diff
@@ -105,9 +106,10 @@ A few important notes about the storage modules:
...
-- Make sure the disks with the storage modules have sufficient available space. The `disk_space` parameter does NOT apply to the storage modules. If you want to create a storage module on a disk smaller than 4 TB, specify a custom size for this module:\
-  `storage_module [number],[size],[mining_address]`\
-  The module will sync data with the weave offsets between `number * size` (in bytes) and `(number + 1) * size`. You also need to give some space to the filesystem, the merkle proofs, and some other metadata stored inside each storage module - about 10% of the configured storage module size.
+- Make sure the disks with the storage modules have sufficient available space for both the data iteself and metadata (10% of the size of the data). Note that `disk_space` command line parameter does NOT apply to the storage modules.
+- If you created storage modules with custom `partition_size` as mentioned above, make sure to specify the `partition_size` in your command line invocation as follow:\
```
Comment (JamesPiechota) [inline mining/mining-guide.md:114]:
```suggestion
- If you created storage modules with custom `storage_module_size` as mentioned above, make sure to specify the `storage_module_size` in your command line invocation as follows:
```

File: mining/mining-guide.md [hunk 5]
```diff
@@ -105,9 +106,10 @@ A few important notes about the storage modules:
...
+  `storage_module [partition_index],[partition_size],[your_mining_address]`\
```
Comment (JamesPiechota) [inline mining/mining-guide.md:115]:
```suggestion
  `storage_module [storage_module_index],[storage_module_size],[your_mining_address]`\
```

File: mining/mining-guide.md [hunk 6]
```diff
@@ -105,9 +106,10 @@ A few important notes about the storage modules:
...
+  The module will sync data with the weave offsets between `partition_index * partition_size` (in bytes) and `(partition_index + 1) * partition_size` at folder `[data_dir]/storage_modules/storage_module[_partition_size]_[partition_index]_[your_mining_address]`.
```
Comment (JamesPiechota) [inline mining/mining-guide.md:116]:
```suggestion
  The module will sync data with the weave offsets between `storage_module_index * storage_module_size` (in bytes) and `(storage_module_index + 1) * storage_module_size` at folder `[data_dir]/storage_modules/storage_module[_storage_module_size]_[storage_module_index]_[your_mining_address]`.
```

**Comment by samantehrani** [comment]
@JamesPiechota Is there a documentation on partition vs storage_module distinction ?
Also is it possible to have storage modules with different sizes ? or should all storage modules be of the same size ?

**Comment by JamesPiechota** [comment]
@samantehrani there isn't any documentation other than in the code unfortunately. The command line help for the `storage_module` flag has a *little* documentation, but not enough.

Youcan use different sizes, unfortunately it gets difficult to make sure the indices line up. So generally you want to use the same size or perhaps sizes that are multiple of each other.

e.g. if you use 900GB and 600GB for your sizes you might have a gap e.g.

sm_0_900GB_addr, and sm_2_600GB_addr

ou would have no data from offset 900GB to 1200GB

So it's possible to use different sizes but it becomes a puzzle if you want to ensure full coverageo of the weave

**Comment by samantehrani** [comment]
Ok, if its not too much trouble, can you point me to the code where I can read more about this enforced partitioning ? is this (fixed) partitioning applied on the indices ? or there is a logical partitioning of data (rather than a linear stacking of data)?

**Comment by JamesPiechota** [comment]
@samantehrani most of it is in the ar_storag_module.erl file you linked to above. There isn't anything forced, it's just how the simple "index + size" storage module scheme plays out. It works just like you described it in your updated docs. The size of the storage module defines how many bytes of the weave it covers, the index or offiset value defines where in the weave the storage module lies. So if the size is 100, and the offset/index is 7, the the storage module covers 700-799

you *can* have overlaps if you specify overlapping storage modules, nothing prevents you from doing that, but unless you want to repack from one address to another it's not recommended (i.e. having overlapping storage modules for the same packing address will not help your hashrate)

**Comment by samantehrani** [comment]
I see, ok so for the sake of Arweave Protocol, StorageModule is the same as Partition (as there is no data partitioning in the protocol). I was just wondering about your comment above "distinction between partition and storage module"

**Comment by JamesPiechota** [comment]
Oh I see what you mean. The protocol only knows about the 3.6TB partitions. A storage module is purely a concept used by the client to provide flexibility to miners - the Arweave protocol has no concept of storage modules. When a miner mines a block the offsets of the chunks being hashes are shared with the network in the block header. The consensus rules used to validate the hashes are defined in terms of 3.6TB partitions and are ignorant of any miner-defined storage modules

**Comment by samantehrani** [comment]
Got it, can you please point me to where this 3.6T enforcement takes place in the protocol code

**Comment by JamesPiechota** [comment]
I'm away from my computer now. But if you search for PARTITION_SIZE you'll see where that value is defined and used. The concept of a partition however is used throughout the code.

**Comment by samantehrani** [comment]
Thanks

**Review by JamesPiechota — APPROVED** [review]

**Pull request merged**